pallets/pallet-bank/src/lib.rs

```diff
@@ -708,4 +757,37 @@ impl<T: Config> Pallet<T> {
 			}
 		})
 	}
+
+	// Return unlock block number.
+	pub fn fund_unlock_at(who: T::AccountId, lock_id: LockId) -> BlockNumberFor<T> {
+		AccountWithUnlockedFund::<T>::iter()
+			.find_map(|(block_number, accounts)| {
```

Comment:
I think find does a map (like a transformation) then a find.
Here we can just use find directly with the map. 
find(|(block, accounts)| accounts.contains)

Reply:
Also add documentations above the new functions